### DISCUSSION TITLE
Add camera (base nodes) to arbitrary objects and make it visible

### DISCUSSION THREAD

**Discussion by tuncalie** [cyberbotics/webots#5291] [General]
I want to add a camera to a pole and also make the camera visible when I view the world. I can add a camera to a robot or a vehicle but the camera itself won't be visible. I could not find a way to add a camera to a pole object. I tried adding it to "slot" but Camera or other base nodes do not appear in the list of possible items to add. Any suggestions on how to achieve these two goals?

**Comment by stefaniapedrazzi** [comment]
A basic `Camera` node has no graphical appearance.
So after adding it to your robot you won't see a mesh representing the camera in the scene.
To make it visible, you can add the desired graphical appearance in the `children` field of the `Camera` node.
Moreover to see the recorded image, you have to enable the camera in the robot controller or in the robot window.
See: https://www.cyberbotics.com/doc/guide/samples-devices#camera-wbt

`Slot` nodes always works with pairs.
When trying to add a camera node into a `Slot` node, the `Camera` basic node is not shown because another `Slot`  node is expected.
Then you can add the Camera or any other node in the `endPoint` field of the second `Slot` node.
https://www.cyberbotics.com/doc/reference/slot

**Reply by tuncalie** [reply]
Thank you, in addition to what you said, I had to first add a Robot object to the endpoint of a Slot and a Camera to the Robot as I wasn't able to add a Camera directly to a Slot.
